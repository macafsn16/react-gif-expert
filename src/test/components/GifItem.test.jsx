import { render, screen } from "@testing-library/react";
import { GifItem } from "../../components";

describe('prieba en GifItem', () => {

    const title = 'Saitama';
    const url = 'https://ono-punch.com/saitama.jgp'
    
  
    test('debe de hacer match con el snapshot ', () => {
      
        const {container} = render( <GifItem title={title} url={url}  /> );

        expect ( container ).toMatchSnapshot();
    });
    

test('debe de mostar la imagen con el URL y el ALT indicado ', () => {

    render( <GifItem title={ title } url={ url }  /> );

   //screen.debug();
    //expect( screen.getByRole('img').src ).toBe(url);
    //expect( screen.getByRole('img').alt ).toBe(title);
   //console.log(screen.getByRole('img'));

   const { src, alt } = screen.getByRole('img');
   expect ( src ).toBe( url );
   expect ( alt ).toBe( alt );

});

test('debe mostar el titulo en el componente ', () => {

    render( <GifItem title={ title } url={ url }  /> );
    expect( screen.getByText(  title) ).toBeTruthy();
  
});




});


