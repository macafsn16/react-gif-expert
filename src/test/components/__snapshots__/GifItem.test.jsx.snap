// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prieba en GifItem debe de hacer match con el snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://ono-punch.com/saitama.jgp"
    />
    <p>
       
      Saitama
       
    </p>
  </div>
</div>
`;
